### PR TITLE
EZEE-3465: Made SiteaccessPreviewVoter repository aware

### PR DESCRIPTION
**Pull request by barw4**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/EZEE-3465
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

As the title states.

Required `ezpublish-kernel` PR: https://github.com/ezsystems/ezpublish-kernel/pull/3085

Regression suite: https://github.com/ezsystems/ezplatform-page-builder/pull/726

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php [hunk 1]
```diff
@@ -43,7 +46,15 @@ public function vote(SiteaccessPreviewVoterContext $context): bool
...
-        if (!in_array($languageCode, $siteaccessLanguages)) {
+        $siteaccessRepository = $this->configResolver->getParameter(
+            'repository',
+            null,
+            $siteaccess
+        );
+        $siteaccessRepository = $siteaccessRepository ?: $this->repositoryConfigurationProvider->pullDefaultRepository();
+        $currentRepository = $this->repositoryConfigurationProvider->getRepositoryConfig()['alias'];
+
+        if (!in_array($languageCode, $siteaccessLanguages) || $siteaccessRepository !== $currentRepository) {
```
Comment (Steveb-p) [inline src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php:57]: We're comparing string vs. string, please use strict check.

We've already discovered a similar issue in validation for choice lists that went undetected.

```suggestion
        if (!in_array($languageCode, $siteaccessLanguages, true) || $siteaccessRepository !== $currentRepository) {
```

Comment (barw4) [reply src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php]: This fragment was here before.

File: src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php [hunk 2]
```diff
@@ -43,7 +46,15 @@ public function vote(SiteaccessPreviewVoterContext $context): bool
...
             $siteaccess
         );
-        if (!in_array($languageCode, $siteaccessLanguages)) {
+        $siteaccessRepository = $this->configResolver->getParameter(
+            'repository',
+            null,
+            $siteaccess
+        );
+        $siteaccessRepository = $siteaccessRepository ?: $this->repositoryConfigurationProvider->pullDefaultRepository();
+        $currentRepository = $this->repositoryConfigurationProvider->getRepositoryConfig()['alias'];
```
Comment (adamwojs) [inline src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php:55]: Could you please move it to private method with proper name in order to improve readability?

**Review by adamwojs — COMMENTED** [review]
Could you please update/provide tests for AbstractSiteaccessPreviewVoter?

**Comment by barw4** [comment]
@adamwojs Tests have been added in https://github.com/ezsystems/ezplatform-admin-ui/pull/1704/commits/d06f819232f04bea86b338ec10a431a1b2dbf9f9

File: src/lib/Tests/Siteaccess/AdminSiteaccessPreviewVoterTest.php [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\Repository\Values\Content\Location;
+use eZ\Publish\Core\Repository\Values\Content\VersionInfo;
+use EzSystems\EzPlatformAdminUi\Siteaccess\AdminSiteaccessPreviewVoter;
+use EzSystems\EzPlatformAdminUi\Siteaccess\SiteaccessPreviewVoterContext;
+use PHPUnit\Framework\TestCase;
+
+class AdminSiteaccessPreviewVoterTest extends TestCase
+{
+    public const LANGUAGE_CODE = 'eng-GB';
```
Comment (adamwojs) [inline src/lib/Tests/Siteaccess/AdminSiteaccessPreviewVoterTest.php:22]: Access level could be private or protected (depends if you designed this class as extendable)

**Review by adamwojs — APPROVED** [review]

File: src/lib/Tests/Siteaccess/AdminSiteaccessPreviewVoterTest.php (hunk 3 above)
Comment (barw4) [reply src/lib/Tests/Siteaccess/AdminSiteaccessPreviewVoterTest.php]: Done in https://github.com/ezsystems/ezplatform-admin-ui/pull/1704/commits/d3314fa6d39765a2764b1bf0e4a18eee480a3575

**Comment by barw4** [comment]
Closed in favor of https://github.com/ezsystems/ezplatform-admin-ui/pull/1706

**Pull request closed without merging**